application/mov/mov_core.go

```diff
@@ -453,6 +454,103 @@ func getTradePairsFromMatchedTx(tx *types.Tx) ([]*common.TradePair, error) {
 	return tradePairs, nil
 }
 
+func tradePairProducer(movStore database.MovStore, tradePairCh chan<- *common.TradePair, closeCh chan struct{}, wg *sync.WaitGroup) {
+	tradePairMap := make(map[string]bool)
+	tradePairIterator := database.NewTradePairIterator(movStore)
+	for tradePairIterator.HasNext() {
+		tradePair := tradePairIterator.Next()
+		if tradePairMap[tradePair.Key()] {
+			continue
+		}
+
+		tradePairMap[tradePair.Key()] = true
+		tradePairMap[tradePair.Reverse().Key()] = true
+
+		select {
+		case <-closeCh:
+			break
```

Comment:
A "break" statement terminates execution of the innermost "for", "switch" or "select" statement.